ArduCopter/Parameters.cpp

```diff
@@ -967,14 +967,46 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
     AP_SUBGROUPINFO(arot, "AROT_", 37, ParametersG2, AC_Autorotation),
 #endif
 
-#if MODE_ZIGZAG_ENABLED == ENABLED && SPRAYER_ENABLED == ENABLED
-    // @Param: ZIGZAG_AUTO_PUMP
-    // @DisplayName: Auto pump in ZigZag
-    // @Description: Enable the auto pump in ZigZag mode. SERVOx_FUNCTION = 22 (SprayerPump) and SPRAY_ENABLE = 1 also must be set. This makes the pump on while moving to destination A or B. The pump will stop if the vehicle reaches destination or the flight mode is changed from ZigZag to other.
+#if MODE_ZIGZAG_ENABLED == ENABLED
+#if SPRAYER_ENABLED == ENABLED
+    // @Param: ZIGZ_SPRAYER
+    // @DisplayName: Auto sprayer in ZigZag
+    // @Description: Enable the auto sprayer in ZigZag mode. SPRAY_ENABLE = 1 and SERVOx_FUNCTION = 22(SprayerPump) / 23(SprayerSpinner) also must be set. This makes the sprayer on while moving to destination A or B. The sprayer will stop if the vehicle reaches destination or the flight mode is changed from ZigZag to other.
     // @Values: 0:Disabled,1:Enabled
     // @User: Advanced
-    AP_GROUPINFO("ZIGZAG_AUTO_PUMP", 38, ParametersG2, zigzag_auto_pump_enabled, ZIGZAG_AUTO_PUMP_ENABLED),
-#endif
+    AP_GROUPINFO("ZIGZ_SPRAYER", 38, ParametersG2, zigzag_spray_enabled, ZIGZAG_SPRAY_ENABLED),
+#endif // SPRAYER_ENABLED == ENABLED
+
+    // @Param: ZIGZ_WP_DELAY
+    // @DisplayName: The delay for zigzag waypoint
+    // @Description: Waiting time after reached the destination
+    // @Units: ms
+    // @Range: 100 10000
+    // @User: Advanced
+    AP_GROUPINFO("ZIGZ_WP_DELAY", 39, ParametersG2, zigzag_wp_delay, ZIGZAG_WP_DELAY),
+
+    // @Param: ZIGZ_SIDE_DIST
+    // @DisplayName: Sideways distance in ZigZag auto
+    // @Description: The distance to move sideways in ZigZag mode
+    // @Units: cm
+    // @Range: 10 10000
+    // @User: Advanced
+    AP_GROUPINFO("ZIGZ_SIDE_DIST", 40, ParametersG2, zigzag_side_dist, 400),
+
+    // @Param: ZIGZ_DIRECTION
+    // @DisplayName: Sideways direction in ZigZag auto
+    // @Description: The direction to move sideways in ZigZag mode
+    // @Values: 0: forward, 1:right, 2: backward, 3: left
+    // @User: Advanced
+    AP_GROUPINFO("ZIGZ_DIRECTION", 41, ParametersG2, zigzag_direction, 0),
+
+    // @Param: ZIGZ_LINE_NUM
+    // @DisplayName: Total number of lines
+    // @Description: Total number of lines for ZigZag auto if 1 or more. -1: Infinity, 0: Disabled
+    // @Range: -1 32767
+    // @User: Advanced
+    AP_GROUPINFO("ZIGZ_LINE_NUM", 42, ParametersG2, zigzag_line_num, 0),
```

Comment:
Would a maximum distance be better?  I'm not sure, just a question..

Reply:
It may be easier for users to input the size of the field..
But I select the total number of lines because I gave -1 and 0 special meaning.